DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/CryptoHashProcessor.cs

```diff
@@ -0,0 +1,91 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text;
+using Dicom;
+using Dicom.IO.Buffer;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class CryptoHashProcessor : IAnonymizerProcessor
+    {
+        private readonly CryptoHashFunction _cryptoHashFunction;
+
+        public CryptoHashProcessor(IDicomAnonymizationSetting ruleSetting = null)
+        {
+             var setting = (DicomCryptoHashSetting)(ruleSetting ?? new DicomCryptoHashSetting());
+             var cryptoHashKey = Encoding.UTF8.GetBytes(setting.CryptoHashKey);
+             _cryptoHashFunction = new CryptoHashFunction(cryptoHashKey);
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            var test = item.GetType().GetMethods();
+            if (dicomDataset.AutoValidate && !IsValidItemForCryptoHash(item))
```

Comment:
shall we depend on dicomDataset.AutoValidate?

Reply:
dicomDataset.AutoValidate is true by default.
If we want to skip validation, we need to set this property as false.

Reply:
Fixed it, thanks.